logging/telemetry_test.go

```diff
@@ -213,6 +214,44 @@ func TestDetails(t *testing.T) {
 	a.Equal(details, data[0]["details"])
 }
 
+func TestHeartbeatDetails(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	a := require.New(t)
+	f := makeTelemetryTestFixture(logrus.InfoLevel)
+
+	var hb telemetryspec.HeartbeatEventDetails
+	hb.Info.Version = "v2"
+	hb.Info.VersionNum = "1234"
+	hb.Info.Channel = "alpha"
+	hb.Info.Branch = "br0"
+	hb.Info.CommitHash = "abcd"
+	hb.Metrics = map[string]float64{
+		"Hello": 38.8,
+	}
+	f.telem.logEvent(f.l, telemetryspec.ApplicationState, telemetryspec.HeartbeatEvent, hb)
+
+	data := f.hookData()
+	a.NotNil(data)
+	a.Len(data, 1)
+	a.Equal(hb, data[0]["details"])
+
+	// assert JSON serialization is backwards compatible
+	js, err := json.Marshal(data[0])
```

Comment:
maybe just store an old serialized value and compare against new?

Reply:
I couldn't figure out how to write a test that had access to the serialized value that the telemetry system makes ... there is a JSON conversion going on inside the logrus hook in a third party library, before being sent via the telemetry client. So I figured I would just simulate that call to json.Marshal here..